api/base/views.py

```diff
@@ -74,7 +74,10 @@ def get_serializer_context(self):
             embeds_partials[embed] = self._get_embed_partial(embed, embed_field)
 
         context.update({
-            'enable_esi': is_truthy(self.request.query_params.get('esi', django_settings.ENABLE_ESI)),
+            'enable_esi': (
+                is_truthy(self.request.query_params.get('esi', django_settings.ENABLE_ESI)) and
+                self.request.accepted_media_type in {'application/vnd.api+json', 'application/json'}
```

Comment:
Would be good to have these media types go in a settings variable--something like `ESI_MEDIA_TYPES`.